### ISSUE TITLE
Неоднозначность автозамены

### ISSUE THREAD

**Issue by Makishima** [BladeMight/Mahou#127]
У меня скачет автозамена "вы ds вы ds"
![mahou](https://user-images.githubusercontent.com/5548003/29816025-582aabbe-8cbb-11e7-803a-c947126b475a.gif)

**Comment by BladeMight** [comment]
Согласен, есть ошибки, файл слишком большой нужно будет долго проверять...

**Comment by Makishima** [comment]
А он c диска каждый раз вычитывается?

**Comment by BladeMight** [comment]
> А он c диска каждый раз вычитывается?

**Только** каждый раз при нажатии на применить.

**Comment by BladeMight** [comment]
@Makishima Обновил словарь, обновись и проверь.

**Comment by Makishima** [comment]
C этим ок. Но есть проблема c постоянным переводом С в латиницу

**Comment by BladeMight** [comment]
Ок, и тут исправил.

**Comment by BladeMight** [comment]
Я вот тут добавил **большие** словари [русский](https://github.com/BladeMight/Mahou/blob/master/Dictionaries-origin/!Russian-big.txt) и [английски](https://github.com/BladeMight/Mahou/blob/master/Dictionaries-origin/!English-big.txt) которые вместе дают около 150 тыс слов для автозамены, и это похоже было слишком много для текущего Mahou-v2.4.5.0(не latest-commit), загружался словарь аж за 8-10 секунд, и плюс тормозил интерфейс при активации вкладки автозамена, а еще это все столько оперативной памяти съедало(доходило аж до 1гб) и процессор 25~50%(на двух-ядерном было бы 50~100%). Ну так к чему это я:
У меня получилось очень сильно улучшить скорость загруски сниппетов ~20-50% и добавить лимит при котором во вкладке автозамена не будет отображаться весь словарь. Сейчас(на версии latest-commit) словарь загружается 4-5 сек. при этом использует до 200мб(может больше если словарь будет больше) оперативной, для загрузки сниппетов, а потом уже 2-4 мб.

Как ты думаешь стоит эти большие словари ставить в AS_dict.txt в репозиторие?

**Comment by Makishima** [comment]
Я за. Однако, до сих пор есть люди с модемным и/или ограниченным интернетом. Возможно ли сделать предупреждение перед загрузкой словаря такого?

**Comment by BladeMight** [comment]
@Makishima Я вот в последней бете улучшил метод загрузки словаря, и теперь загружает 150000+ слов за **менее чем 1 секунду**(убрал regex и написал собственный парсер). А еще я добавлю возможность выбирать как загружать словарь [zip](https://github.com/BladeMight/Mahou/releases/download/latest-commit/AS_dict.zip) или [txt](https://raw.githubusercontent.com/BladeMight/Mahou/master/AS_dict.txt). Еще можно добавить чтоб при первом нажатии на кнопку(обновить словарь) отображался размер, а при втором скачивалось.

**Comment by BladeMight** [comment]
@Makishima Сделал все о чем писал, проверь.

**Comment by Makishima** [comment]
Не в ближайшую пару недель :)

**Comment by BladeMight** [comment]
@Makishima Ну не страшно, подожду... :)

**Comment by chatmag** [comment]
Версия 2,5,0,1 - слово "еще" конвертировало в английский tot. В AS_dict.txt (6.27) метров именно такого слова нет, вышел из положения, убрав из словаря все слова, начинающиеся на "еще" (tot)

**Comment by BladeMight** [comment]
@chatmag Все из-за того что в релиз версии 2.5.0.1 не игнорируются закомментированные сниппеты, а в latest-commit все уже исправлено, сейчас сделал релиз 2.5.0.5, обновитесь.
И еще может стоит обновить словарь.
